kubectl-openebs/cli/command/cstor/pool/pool_list.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020-2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pool
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/openebs/openebsctl/client"
+	"github.com/openebs/openebsctl/kubectl-openebs/cli/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	poolListCommandHelpText = `
+	This command lists of all known pools in the Cluster.
+	
+	Usage: kubectl openebs cStor pool list [options]
+	`
+	namespace string
+)
+
+// NewCmdPoolsList displays status of OpenEBS Pool(s)
+func NewCmdPoolsList() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Displays status information about Pool(s)",
+		Long:  poolListCommandHelpText,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(RunPoolsList(cmd), util.Fatal)
+		},
+	}
+
+	cmd.Flags().StringVarP(&namespace, "namespace", "n", "openebs",
+		"namespace name, required if pool is not in the `default` namespace")
+
+	flag.CommandLine.Parse([]string{})
+
+	return cmd
+}
+
+//RunPoolsList fetchs & lists the pools
+func RunPoolsList(cmd *cobra.Command) error {
+
+	client, err := client.NewK8sClient(namespace)
+	util.CheckErr(err, util.Fatal)
+
+	cpools, err := client.GetcStorPools()
```

Comment:
Command says list about cStor pools but we are listing CStorPoolClusters... Is that expected?

Reply:
- cStor pools are denoted by CSPCs as far as I know. If we are talking about listing both CSPCs and SPCs then we are listing only CSPCs as moving towards the latest versions of OpenEBS SPCs would be deprecated.

Reply:
@mittachaitu Did you mean pools as in pool-instances and not pool clusters?

Reply:
Yes, I should have specified the term `cstorPoolInstance` in the earlier comments.

Actually, In cStor there are more confusing terms(this will get clear off when one started using/working on that). What I usually think(Not sure whether it is valid or not):
cStor pools :-  Actual underlying pools on disk (User no need to worry about this).
Pool Instances/cStor Pool Instances:- Abstraction/representation of underlying disk pools(Represents various stats of underlying pool).
cStorpoolcluster/cStor cluster:- Represents a group of pool instances.

Basically, CLI sub-commands will depends on what we have to project.

Reply:
@mittachaitu In the subsequent PRs, would were planning to add a describe feature which takes a `cspc` name and displays all sorts of information, related to its `instances`. 
Would that be more meaningful or shall we list the instances in this `list` command and for describe we would take the name of the instance and display details related to it. 
These are two scenarios we can have. Please let us know which suits the purpose more.

Reply:
Here, when the describe of CSPC command is triggered and if it displays info about instances makes sense(since motive is to get enough information by running single command). 

IMO we can make it more simple and clear in the following way(**_Not sure how far below pattern is valid_**):

- Displaying all the CSPCs and their information present in the cluster
```sh
kubectl openebs get cStor poolcluster
Name           Namespace  HealthyInstances  ProvisionedInstances  DesiredInstances  Age
----           ---------  ----------------  --------------------  ----------------  ---
cstor-storage-1  openebs    1                 1                     1                 3d27m
cstor-storage-2  openebs    1                 1                     1                 3d27m
cstor-storage-3  openebs    1                 1                     1                 3d27m
```

- Displaying the CSPIs information when a user specifies CSPC name along with the command (I not sure whether it is valid or not). 
```sh
kubectl openebs get cStor poolcluster <cstor-storage-1>
Name           Namespace  UsedSpace  AvailableSpace TotalSpace TotalReplicas HealthyReplicas  Age
----           ---------  ----------------  --------------------  ----------------  ---
...
...
...
```
- Above pattern will avoid introducing other sub-commands like poolinstance.

Reply:
- Displaying all the CSPCs and their information present in the cluster
   For the above we use the verb `list` in place of `get`.

- Displaying the CSPIs information when a user specifies CSPC name along with the command 
  For the above we would use the verb `describe` in place of `get`. 

- We will change the term from `pool` to `poolcluster`

@mittachaitu If these seems fine we can finalize on this.

Reply:
Make sense :+1:

Reply:
Thank you.

Reply:
If we are listing `CSPI` then use `pool`, if using `CSPC` then using PoolCluster .. makes sense.. but let us avoid using Pool Cluster -- as it not a commonly understood term.

Reply:
- We were listing `CSPCs` initially, as `CSPCs` are a group of instances, which upon description would have brought out the underlying `CSPIs`, but seems like `CSPIs` would be the requirement as a list. I will keep the command same i.e. `kubectl openebs cStor pool list` and change the output from `CSPCs` to `CSPIs`.

Reply:
- I have updated the PR description, with the latest output of this command.